### PR TITLE
refactor: peer report

### LINKED ISSUE #1561: Move `PeerReportBehavior` functionality to `FuelBehavior`
In our upgrading of `libp2p` we moved `identify` and `heartbeat` out of `PeerReportBehavior` and into the `FuelBehavior`. There is still some functionality left in the `PeerReportBehavior`, however that probably can be moved to `FuelBehavior`.

We can just get rid of `PeerReport` all together, probably :+1:

### PR DESCRIPTION
**Pull request by matt-user**
[Link to related issue(s) here, if any]
closes #1561 

[Short description of the changes.]

## Checklist
- [ ] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests
- [ ] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [ ] I have reviewed the code myself
- [ ] I have created follow-up issues caused by this PR and linked them here

### After merging, notify other teams

[Add or remove entries as needed]

- [ ] [Rust SDK](https://github.com/FuelLabs/fuels-rs/)
- [ ] [Sway compiler](https://github.com/FuelLabs/sway/)
- [ ] [Platform documentation](https://github.com/FuelLabs/devrel-requests/issues/new?assignees=&labels=new+request&projects=&template=NEW-REQUEST.yml&title=%5BRequest%5D%3A+) (for out-of-organization contributors, the person merging the PR will do this)
- [ ] Someone else?

### CODE REVIEW

File: crates/services/p2p/src/decay.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+const REPUTATION_DECAY_INTERVAL_IN_SECONDS: u64 = 1;
+
+#[derive(Debug, Clone)]
+pub enum DecayEvent {
+    /// Informs p2p service / PeerManager to perform reputation decay of connected nodes
+    PerformDecay,
+}
+
+pub struct Behavior {
+    decay_interval: Interval,
```
Comment (MitchTurner) [inline crates/services/p2p/src/decay.rs:38]: Why can't we just put `decay_interval` on to `FuelBehaviour`?

File: crates/services/p2p/src/health_check.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+const HEALTH_CHECK_INTERVAL_IN_SECONDS: u64 = 10;
+
+#[derive(Debug, Clone)]
+pub enum HealthCheckEvent {
+    /// Informs p2p service / PeerManager to check health of reserved nodes' connections
+    CheckReservedNodesHealth,
+}
+
+pub struct Behavior {
+    health_check: Interval,
```
Comment (MitchTurner) [inline crates/services/p2p/src/health_check.rs:38]: Same question here.

File: crates/services/p2p/src/decay.rs (hunk 1 above)
Comment (matt-user) [reply crates/services/p2p/src/decay.rs:38]: in order for the #[derive(NetworkBehaviour] macro to work all fields of the structs must impl Network behaviour

Comment (MitchTurner) [reply crates/services/p2p/src/decay.rs:38]: I see. I'm not really sure what the purpose of this issue was then tbh. I don't feel like moving each part to separate behaviors is buying us much.

Comment (matt-user) [reply crates/services/p2p/src/decay.rs:38]: makes sense I can close the issue

Comment (MitchTurner) [reply crates/services/p2p/src/decay.rs:38]: I'd just leave it for now. Maybe someone else remembers the context. It's possible to implement `NetworkBehavior` manually, so maybe that's the play? but it's low priority anyway.

**Comment by matt-user** [comment]
closing see comment here https://github.com/FuelLabs/fuel-core/pull/1944#discussion_r1630302798

**Pull request closed without merging**